File: app/src/main/java/ru/aradxxx/ciceronetabssample/di/NavigationModule.java
package ru.aradxxx.ciceronetabssample.di;

import javax.inject.Singleton;

import androidx.annotation.NonNull;
import dagger.Module;
import dagger.Provides;
import ru.aradxxx.ciceronetabs.TabCicerone;
import ru.aradxxx.ciceronetabs.TabRouter;
import ru.aradxxx.ciceronetabs.TabRouterFactoryImpl;

@Module
public class NavigationModule {
    @Provides
    @Singleton
    @NonNull
    TabCicerone<TabRouter> provideTabRouterHolder() {
        return new TabCicerone<>(new TabRouterFactoryImpl());
    }
}

File: app/src/main/java/com/github/aradxxx/ciceroneflowssample/di/NavigationModule.kt
package com.github.aradxxx.ciceroneflowssample.di

import com.github.aradxxx.ciceroneflow.DefaultFlowRouterFactory
import com.github.aradxxx.ciceroneflow.FlowCicerone
import com.github.aradxxx.ciceroneflow.FlowRouter
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class NavigationModule {
    @Provides
    @Singleton
    fun provideTabRouterHolder(): FlowCicerone<FlowRouter> {
        return FlowCicerone(DefaultFlowRouterFactory())
    }
}
